maven/datasets/general_election/uk_2010_results.py

```diff
@@ -20,128 +20,198 @@
+        #
+        results.columns = [
+            self.parties_lookup[x] if x in self.parties_lookup else x
+            for x in results.columns
+        ]
+
+        # Calculate constituency level vote share proportion (pc=proportion count)
+        for party in self.parties_lookup.values():
+            results[party + "_pc"] = results[party] / results["Votes"]
 
         # Create PANO -> geo lookup
-        results['geo'] = results.Region.map({
-            'East Midlands': 'England_not_london',
-            'Eastern': 'England_not_london',
-            'London': 'London',
-            'North East': 'England_not_london',
-            'North West': 'England_not_london',
-            'Northern Ireland': 'NI',
-            'Scotland': 'Scotland',
-            'South East': 'England_not_london',
-            'South West': 'England_not_london',
-            'Wales': 'Wales',
-            'West Midlands': 'England_not_london',
-            'Yorkshire and the Humber': 'England_not_london',
-            })
-        assert results.loc[237.0, 'geo'] == 'London'
-
-        # Calculate geo-level vote share
-        # TODO: Do we use this?
-        results_by_geo = results.loc[:, ['Votes', 'geo'] + list(parties_lookup.values())].groupby('geo').sum()
-        results_by_geo_voteshare = results_by_geo.div(results_by_geo['Votes'], axis=0)
-        del results_by_geo_voteshare['Votes']
+        # Store regions in England as either London or not-London, rather than
+        # having East/West Midlands and such.
+        results["geo"] = results.Region.map(
+            self.uk_regions_to_NI_scotland_london_and_not_london
+        )
+
+        assert results.loc[237.0, "geo"] == "London"
 
         # Who won?
         def winner(row):
-            all_parties = set(results_full.columns[6:]) - set(['Other'])
-            winning_party = row[all_parties].sort_values(ascending=False).index[0]
-            if winning_party in parties_lookup.keys():
-                winning_party = parties_lookup[winning_party]
-            elif winning_party == 'Speaker':
-                winning_party = 'other'
-            return winning_party
-
-        results['winner'] = results_full.apply(winner, axis=1)
-        # Check Labour won 306 seats in 2010.
-        assert results.groupby('winner').count()['Constituency Name'].sort_values(ascending=False)[0] == 306
+            """
+            given a row representing outome for a consituency sort and return
+            the winning party
+            """
+            winning_party = (
+                row[self.first_col_with_party_vote_info :]
+                .sort_values(ascending=False)
+                .index[0]
+            )
+            if winning_party in self.parties_lookup.keys():
+                winning_party = self.parties_lookup[winning_party]
+            elif winning_party == "Speaker":
+                winning_party = "other"
+            return winning_party.lower()
```

Comment:
Do we want this .lower() here? Without it the downloaded and processed data is exactly the same as master.

(I used my URL fixes from #15 to make this test)

Reply:
Yeah this can go